cli/src/pcluster/config/cluster_config.py

```diff
@@ -971,6 +970,27 @@ def _register_validators(self, context: ValidatorContext = None):  # noqa: D102
             self._register_validator(InstanceProfileValidator, instance_profile_arn=self.instance_profile)
 
 
+class HeadNodeAndQueueIam(_BaseIam):
+    """Represent the IAM configuration for HeadNode and Queue."""
+
+    def __init__(
+        self,
+        s3_access: List[S3Access] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.s3_access = s3_access
+
+
+class LoginNodeIam(_BaseIam):
+    """Represent the IAM configuration for HeadNode and Queue."""
```

Comment:
`... IAM configuration for login node.`

Reply:
Add/Extend a unit test to confirm that the instance profile and instance role cannot be used together.

Reply:
Done!